packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,592 @@
+  }
+
+  /**
+   * allows developer to provide item path for a file
+   *
+   * @type {string}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'item-path'
+  })
+  public get itemPath(): string {
+    return this._itemPath;
+  }
+  public set itemPath(value: string) {
+    if (value === this._itemPath) {
+      return;
+    }
+
+    this._itemPath = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide user id for a file
+   *
+   * @type {string}
+   * @memberof MgtFile
+   */
+  @property({
+    attribute: 'user-id'
+  })
+  public get userId(): string {
+    return this._userId;
+  }
+  public set userId(value: string) {
+    if (value === this._userId) {
+      return;
+    }
+
+    this._userId = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide insight type for a file
+   * can be trending, used, or shared
+   *
+   * @type {OfficeGraphInsightString}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'insight-type'
+  })
+  public get insightType(): OfficeGraphInsightString {
+    return this._insightType;
+  }
+  public set insightType(value: OfficeGraphInsightString) {
+    if (value === this._insightType) {
+      return;
+    }
+
+    this._insightType = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * Sets what data to be rendered (file icon only, oneLine, twoLines threeLines).
+   * Default is 'threeLines'.
+   *
+   * @type {ViewType}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'view',
+    converter: value => {
+      if (!value || value.length === 0) {
+        return ViewType.threelines;
+      }
+
+      value = value.toLowerCase();
+
+      if (typeof ViewType[value] === 'undefined') {
+        return ViewType.threelines;
+      } else {
+        return ViewType[value];
+      }
+    }
+  })
+  public view: ViewType;
+
+  /**
+   * allows developer to provide file type to filter the list
+   * can be docx
+   *
+   * @type {string[]}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'file-extensions',
+    converter: (value, type) => {
+      return value.split(',').map(v => v.trim());
+    }
+  })
+  public get fileExtensions(): string[] {
+    return this._fileExtensions;
+  }
+  public set fileExtensions(value: string[]) {
+    if (arraysAreEqual(this._fileExtensions, value)) {
+      return;
+    }
+
+    this._fileExtensions = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * A number value to indicate the maximum number of files to show
+   * @type {number}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'show-max',
+    type: Number
+  })
+  public showMax: number;
+
+  /**
+   * A number value to indicate the number of more files to load when show more button is clicked
+   * @type {number}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'load-more-file-count',
+    type: Number
+  })
+  public loadMoreFileCount: number;
```

Comment:
Consider renaming this to `pageSize` (`page-size`)

Reply:
I love it. I couldn't think of a good name haha.